oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticBulkProcessorHandler.java

```diff
@@ -206,6 +222,9 @@ public boolean close() throws IOException {
             }
         }
 
+        // TODO: workaround for https://github.com/elastic/elasticsearch-java/pull/867 remove when fixed
+        scheduler.shutdownNow();
```

Comment:
The thread pool is not shutdown if the `close` method returns or fails before this point. There is a return statement above. And if any exceptions are thrown, then this line is also skipped. 
Also consider using this class to shutdown the executor gracefully: https://github.com/apache/jackrabbit-oak/blob/trunk/oak-commons/src/main/java/org/apache/jackrabbit/oak/commons/concurrent/ExecutorCloser.java